### PR TITLE
Object bounds refactor

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

For @walkawayy's #1267

### CODE REVIEW

**Comment by walkawayy** [comment]
Bad news. I couldn't anything to work. Nothing happens when trying to use pickups, keyhole, puzzle hole, midas hand, Scion altar, etc,

**Comment by rr-** [comment]
> Bad news. I couldn't anything to work. Nothing happens when trying to use pickups, keyhole, puzzle hole, midas hand, Scion altar, etc,

Sorry, dumb mistake in Item_TestPosition (`y < min || y > min` instead of `y < min || y > max`), corrected now.

File: src/game/objects/traps/movable_block.c [hunk 1]
```diff
@@ -22,28 +22,30 @@ typedef enum {
...
-    -10 * PHD_DEGREE,
-    +10 * PHD_DEGREE,
+static const OBJECT_BOUNDS m_MovableBlock_Bounds = {
+    .shift = {
+        .min = { .x = -300, .y = 0, .z = -WALL_L / 2 - (LARA_RAD + 80), },
+        .max = { .x = +300, .y = 0, .z = -WALL_L / 2, },
+    },
+    .rot = {
+        .min = { .x = +10 * PHD_DEGREE, .y = +30 * PHD_DEGREE, .z = +10 * PHD_DEGREE, },
+        .max = { .x = -10 * PHD_DEGREE, .y = -30 * PHD_DEGREE, .z = -10 * PHD_DEGREE, },
```
Comment (walkawayy) [inline src/game/objects/traps/movable_block.c]: I can't move movable blocks. I think these min and max are reversed?

**Review by walkawayy — CHANGES_REQUESTED** [review]
Movable blocks, the scion altar, and underwater pickups seem to be broken. I had walk to items on as well fwiw.

File: src/game/objects/traps/movable_block.c (hunk 1 above)
Comment (rr-) [reply src/game/objects/traps/movable_block.c]: Good catch, I've fixed this.

**Comment by rr-** [comment]
@walkawayy Thank you for testing. I've fixed Scion and UW pickups as well.

**Review by lahm86 — CHANGES_REQUESTED** [review]
There is only one issue that I found, otherwise this all looks good.

`Scion_Setup2` in `scion.c` needs `g_Objects[O_SCION_ITEM2].bounds = Pickup_Bounds;` and that function being made public (or something similar). This is the scion that Pierre drops. Currently it crashes when his items spawn.

For clarity, I checked all of the following, both with and without walk-to-items enabled:
- Regular pickups
- Carried item pickups
- Underwater pickups
- Keyholes
- Puzzle slots
- Levers
- Underwater levers
- Save crystals
- Midas hand
- Pushblocks
- Qualopec scion
- Sanctuary scion
- Atlantis scion
- Pyramid scion - just to test it can't be picked up :)

**Comment by rr-** [comment]
Thank you – adjusted.

**Review by lahm86 — APPROVED** [review]

**Pull request merged**